keywords/MobileRestClient.py

```diff
@@ -608,11 +603,20 @@ def add_doc(self, url, db, doc, auth=None):
         doc["updates"] = 0
 
         if auth_type == AuthType.session:
-            resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), cookies=dict(SyncGatewaySession=auth[1]))
+            if use_post:
+                resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), cookies=dict(SyncGatewaySession=auth[1]))
+            else:
+                resp = self._session.put("{}/{}/{}".format(url, db, doc["_id"]), data=json.dumps(doc), cookies=dict(SyncGatewaySession=auth[1]))
         elif auth_type == AuthType.http_basic:
-            resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), auth=auth)
+            if use_post:
+                resp = self._session.post("{}/{}/".format(url, db), data=json.dumps(doc), auth=auth)
```

Comment:
I think this could reduce duplication as:

```
if use_post:
    op = self._session.post
else:
    op = self._session.put
op("{}/{}/".format(url, db), data=json.dumps(doc), auth=auth)
```